test/make.sh

```diff
@@ -14,87 +14,146 @@
+    compat=("--no-run-if-empty")
+  fi
+  xargs "${compat[@]}" "$@"
 }
 
 # This function makes sure that the required files for
 # releasing to OSS are present
 function basefiles() {
-  echo "Checking for required files"
-  test -f LICENSE || echo "Missing LICENSE"
-  test -f README.md || echo "Missing README.md"
+  local fn required_files="LICENSE README.md"
+  echo "Checking for required files ${required_files}"
+  for fn in ${required_files}; do
+    test -f "${fn}" || echo "Missing required file ${fn}"
+  done
 }
 
 # This function runs the hadolint linter on
 # every file named 'Dockerfile'
 function docker() {
   echo "Running hadolint on Dockerfiles"
-  find . -name "Dockerfile" -exec hadolint {} \;
+  find_files . -name "Dockerfile" -print0 \
+    | compat_xargs -0 hadolint
 }
 
 # This function runs 'terraform validate' against all
-# files ending in '.tf'
+# directory paths which contain *.tf files.
 function check_terraform() {
   echo "Running terraform validate"
-  #shellcheck disable=SC2156
-  find . -name "*.tf" -not -path "./autogen/*" -not -path "./test/fixtures/shared/*" -not -path "./test/fixtures/all_examples/*" -exec bash -c 'terraform validate --check-variables=false $(dirname "{}")' \;
-  echo "Running terraform fmt"
-  #shellcheck disable=SC2156
-  find . -name "*.tf" -not -path "./autogen/*" -not -path "./test/fixtures/shared/*" -not -path "./test/fixtures/all_examples/*" -exec bash -c 'terraform fmt -check=true -write=false "{}"' \;
+  find_files . -name "*.tf" -print0 \
+    | compat_xargs -0 -n1 dirname \
+    | sort -u \
+    | grep -xv './test/fixtures/shared' \
+    | compat_xargs -t -n1 terraform validate --check-variables=false
 }
 
 # This function runs 'go fmt' and 'go vet' on every file
 # that ends in '.go'
 function golang() {
   echo "Running go fmt and go vet"
-  find . -name "*.go" -exec go fmt {} \;
-  find . -name "*.go" -exec go vet {} \;
+  find_files . -name "*.go" -print0 | compat_xargs -0 -n1 go fmt
+  find_files . -name "*.go" -print0 | compat_xargs -0 -n1 go vet
 }
 
 # This function runs the flake8 linter on every file
 # ending in '.py'
 function check_python() {
   echo "Running flake8"
-  find . -name "*.py" -exec flake8 {} \;
+  find_files . -name "*.py" -print0 | compat_xargs -0 flake8
+  return 0
 }
 
 # This function runs the shellcheck linter on every
 # file ending in '.sh'
 function check_shell() {
   echo "Running shellcheck"
-  find . -name "*.sh" -exec shellcheck -x {} \;
+  find_files . -name "*.sh" -print0 | compat_xargs -0 shellcheck -x
 }
 
 # This function makes sure that there is no trailing whitespace
 # in any files in the project.
 # There are some exclusions
 function check_trailing_whitespace() {
-  echo "The following lines have trailing whitespace"
-  grep -r '[[:blank:]]$' --exclude-dir=".terraform" --exclude-dir=".kitchen" --exclude="*.png" --exclude="*.pyc" --exclude-dir=".git" .
+  local rc
+  echo "Checking for trailing whitespace"
+  find_files . -print \
```

Comment:
Seems to me that it might make sense for this to scan `autogen` as well -- if a file created by template generation has trailing whitespace and its source template does as well, shouldn't we be flagging the source template?

Reply:
Although I also can see the argument for maintaining consistency and only flagging generated files. Happy to discuss this -- I don't have a particularly strong preference. Perhaps an appropriate middle ground might be checking for a source template and alerting that it might be the underlying source of the issue? If we do that here, though, we might want to also do it in other checks.